tests/XAMLMarkupExtensions.UnitTests/ObjectDependencyManagerTests.cs

```diff
@@ -123,24 +131,109 @@ public void AddObjectDependency_DependencyAlive_TargetAlive()
+            CreateWeakReference(out var secondDependency, out var secondDependencyWeakReference);
+
+            // ACT.
+            // Register targets and its dependencies.
+            var isFirstRegistered = ObjectDependencyManager.AddObjectDependency(firstDependencyWeakReference, firstTarget);
+            var isSecondRegistered = ObjectDependencyManager.AddObjectDependency(secondDependencyWeakReference, secondTarget);
+            
+            // Remove direct references to targets.
+            firstTarget = null;
+            secondTarget = null;
+            
+            // Force collect. ObjectDependencyManager should has direct references to targets, so they will not collected.
+            GC.Collect();
+
+            // ASSERT.
+            Assert.True(isFirstRegistered);
+            Assert.True(isSecondRegistered);
+            Assert.True(firstTargetWeakReference.IsAlive);
+            Assert.True(secondTargetWeakReference.IsAlive);
+            
+            // Prevent references optimization.
+            Assert.NotNull(firstDependency);
+            Assert.NotNull(secondDependency);
+        }
+        
+        /// <summary>
+        /// Check that AddObjectDependency method calls CleanUp inside and remove first target if its dependencies are dead.
+        /// </summary>
+        [Fact]
+        public void AddObjectDependency_TwoTargetsFirstIsDead_FirstTargetCleaned()
+        {
+            // ARRANGE.
+            CreateWeakReference(out var firstTarget, out var firstTargetWeakReference);
+            CreateWeakReference(out var firstDependency, out var firstDependencyWeakReference);
+            CreateWeakReference(out var secondTarget, out var secondTargetWeakReference);
+            CreateWeakReference(out var secondDependency, out var secondDependencyWeakReference);
+
+            // ACT.
+            // Register first target and its dependency.
+            var isFirstRegistered = ObjectDependencyManager.AddObjectDependency(firstDependencyWeakReference, firstTarget);
+            
+            // Remove direct references to first target and to the dependency.
+            firstTarget = null;
+            firstDependency = null;
+            
+            // Force collect first dependency.
+            GC.Collect();
+            
+            // First dependency is collected.
+            Assert.False(firstDependencyWeakReference.IsAlive);
+            
+            // First target should be still alive, because ObjectDependencyManager doesn't know that dependency is dead.
+            Assert.True(firstTargetWeakReference.IsAlive);
+            
+            // Register second target and its dependency. After that should cleaned first target. 
+            var isSecondRegistered = ObjectDependencyManager.AddObjectDependency(secondDependencyWeakReference, secondTarget);
+            
+            // Force collect first target.
+            GC.Collect();
```

Comment:
@Karnah I'm not shure that an simple GC.Collect always enforce the full delete.

Reply:
GC.Collect() collect all generations and use Default collection mode (which equals to GCCollectionMode.Forced).
There is no objects with finalizator, so it seems to me that it's really work.